core/src/main/scala/cats/MonadState.scala

```diff
@@ -17,6 +17,9 @@ package cats
  * }}}
  */
 trait MonadState[F[_], S] extends Monad[F] {
+  def state[A](f: S => (S, A)): F[A] =
+    flatMap(get)(s => f(s) match { case (s, a) => inspect(_ => a)})
```

Comment:
Shouldn't this use `set` to set the `S` resulting from `f(s)`? As is this will just return the initial `S`, right?